File: java/src/main/java/S012.java
public class S012 {
    public static long solve() {
        int index = 1;
        while (true) {
            if (Maths.numDivisors(Maths.triangleNumber(index)) > 500) {
                return Maths.triangleNumber(index);
            }
            index++;
        }
    }
}

File: cpp/S012.cpp
#include <iostream>
#include "maths.cpp"

using namespace std;

static const int TARGET = 500;

int main() {
	int idx = 1;
	while (true) {
		int tri = triangle(idx++);
		if (num_divisors(tri) > TARGET) {
			cout << tri << endl;
			return 0;
		}
	}
}
